x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/details/index.tsx

```diff
@@ -147,7 +158,7 @@ export const RuleDetailsComponent = memo<RuleDetailsComponentProps>(({ signalsIn
                               <EuiButton
                                 href={`#${DETECTION_ENGINE_PAGE_NAME}/rules/id/${ruleId}/edit`}
                                 iconType="visControls"
-                                isDisabled={rule?.immutable ?? true}
+                                isDisabled={!canUserCRUD || (rule?.immutable ?? true)}
```

Comment:
Your types look correct but just a heads up that on the backend we are always returning `immutable` for both `true` and `false` and never omitting it so you should be able to when you want to change your type here and reduce the amount of decision logic in your code.

If for some reason `immutable` is not being returned please let us know.

Reply:
ok I will fix the io-ts so we do not have that

Reply:
ok this is not about the immutable property, it is because when I am fetching the data for rule, the rule object will start to be null and then I want to disable it for security reasons.

Reply:
Got it thanks!